fe/fe-core/src/main/java/org/apache/doris/datasource/hive/event/MetastoreEventFactory.java

```diff
@@ -68,19 +73,71 @@ public List<MetastoreEvent> transferNotificationEventToMetastoreEvents(Notificat
 
     List<MetastoreEvent> getMetastoreEvents(List<NotificationEvent> events, HMSExternalCatalog hmsExternalCatalog) {
         List<MetastoreEvent> metastoreEvents = Lists.newArrayList();
+        String catalogName = hmsExternalCatalog.getName();
         for (NotificationEvent event : events) {
-            metastoreEvents.addAll(transferNotificationEventToMetastoreEvents(event, hmsExternalCatalog.getName()));
+            metastoreEvents.addAll(transferNotificationEventToMetastoreEvents(event, catalogName));
         }
-        return createBatchEvents(metastoreEvents);
+        return createBatchEvents(catalogName, metastoreEvents);
     }
 
     /**
-     * Create batch event tasks according to HivePartitionName to facilitate subsequent parallel processing.
-     * For ADD_PARTITION and DROP_PARTITION, we directly override any events before that partition.
-     * For a partition, it is meaningless to process any events before the drop partition.
-     */
-    List<MetastoreEvent> createBatchEvents(List<MetastoreEvent> events) {
-        // now do nothing
-        return events;
+     * Merge events to reduce the cost time on event processing, currently mainly handles MetastoreTableEvent
+     * because merge MetastoreTableEvent is simple and cost-effective.
```

Comment:
Some examples are given in the comments, such as: if there are both drop table events and add partition events, the add partition event will be ignored